python/sglang/srt/model_executor/model_runner.py

```diff
@@ -556,18 +559,18 @@ def forward_decode(self, forward_batch: ForwardBatch):
         )
 
     def forward_extend(self, forward_batch: ForwardBatch):
-        if self.is_generation:
-            return self.model.forward(
-                forward_batch.input_ids, forward_batch.positions, forward_batch
-            )
-        else:
+        if self.is_embedding:
             # Only embedding models have get_embedding parameter
             return self.model.forward(
                 forward_batch.input_ids,
                 forward_batch.positions,
                 forward_batch,
                 get_embedding=True,
             )
+        else:
```

Comment:
Why do you reverse this condition? Ideally, we want the common case to be the first branch. Could you put the generation case as the first branch?

Reply:
I reversed it because it felt like keeping the `if self.is_embedding` condition close to the embedding-specific functionality made the code a bit more readable. But I have switched it back now!